src/MOI_wrapper/nonlinear_constraints.jl

```diff
@@ -0,0 +1,124 @@
+# Nonlinear constraints (objective not supported)
+
+MOI.supports(::Optimizer, ::MOI.NLPBlock) = true
+
+function MOI.set(o::Optimizer, ::MOI.NLPBlock, data::MOI.NLPBlockData)
+    # We don't actually store the data (to be queried later during the solve
+    # process). Instead, we extract the expression graphs and add the
+    # corresponding constraints to the model directly.
+    if data.has_objective
+        @warn "Nonlinear objective not supported by SCIP.jl!" maxlog=1
+    end
+
+    MOI.initialize(data.evaluator, [:ExprGraph])
+    for i in 1:length(data.constraint_bounds)
+        expr = MOI.constraint_expr(data.evaluator, i)
+        bounds = data.constraint_bounds[i]
+        cr = add_constraint_from_expr(o, expr, bounds.lower, bounds.upper)
+        # not registering or returning constraint reference!
+    end
+
+    return nothing
+end
+
+# Local methods.
+
+const OPMAP = Dict{Symbol, SCIP_ExprOp}(
+    :+ => SCIP_EXPR_SUM,
+    :* => SCIP_EXPR_PRODUCT,
+    :- => SCIP_EXPR_MINUS,
+    :/ => SCIP_EXPR_DIV,
+    :^ => SCIP_EXPR_REALPOWER,
+    :sqrt => SCIP_EXPR_SQRT,
+    :exp => SCIP_EXPR_EXP,
+    :log => SCIP_EXPR_LOG,
+)
+
+mutable struct NonlinExpr
```

Comment:
Could you add a few comments on this data structure?

Reply:
Good idea, especially with the abundance of special cases in the translation.

Reply:
I refactored to code instead, to not use this data structure anymore.